lib/Sema/SemaExpr.cpp

```diff
@@ -11935,7 +11935,20 @@ ExprResult Sema::ActOnCountBoundsExpr(SourceLocation BoundsKWLoc,
                                       CountBoundsExpr::Kind Kind,
                                       Expr *CountExpr,
                                       SourceLocation RParenLoc) {
-  return new (Context) CountBoundsExpr(Kind, CountExpr, BoundsKWLoc, RParenLoc);
+  ExprResult Result = UsualUnaryConversions(CountExpr);
```

Comment:
This function does the "usual C integer conversions".   If If the type is a character, a short integer, a bit field, or an enumeration type, the expression is promoted to an \keyword{int} type if that is large enough to  hold all values of the type or an unsigned int type otherwise.